packages/react/src/auto/hooks/useRelatedModelManager.tsx

```diff
@@ -0,0 +1,39 @@
+import { useApi } from "../../GadgetProvider.js";
+import { GadgetHasManyConfig } from "../../internal/gql/graphql.js";
+import { getModelManager } from "../../utils.js";
+import { useFieldMetadata } from "./useFieldMetadata.js";
+
+export const useRelatedModelManager = (props: {
+  field: string; // Field API identifier
+}) => {
+  const { field } = props;
+
+  const { path, metadata } = useFieldMetadata(field);
+  const config = metadata.configuration as GadgetHasManyConfig;
+
+  if (!config || !config.relatedModel) {
+    throw new Error(`Related model for field "${field}" not found in metadata`);
+  }
+  const apiIdentifier = config.relatedModel.apiIdentifier;
+  const namespace = config.relatedModel.namespace;
+  const inverseFieldApiIdentifier = config.inverseField?.apiIdentifier;
+
+  const api = useApi();
+  const modelManager = getModelManager(api, apiIdentifier, namespace);
+
+  if (!modelManager) {
+    const relatedModelNamespacedApiId = [...(config.relatedModel.namespace ?? []), config.relatedModel.apiIdentifier].join(".");
+    throw new Error(`no model manager for "${relatedModelNamespacedApiId}" found for action function`);
+  }
+
+  return {
+    path,
+    metadata,
+    modelManager,
+    model: {
+      apiIdentifier,
+      namespace,
+      inverseFieldApiIdentifier,
+    },
+  };
```

Comment:
This is just a style thing but it seems weird that this hook returns all this extra stuff. I think it might be a little easier to understand if it just returned the model manager (do one thing and one thing only)

Reply:
Good call. I've replaced the `spread and forward them spread` style with a `spread locally, forward the unspread` style when possible. Definitely helps to reduce the number of forwarded things